### DISCUSSION TITLE
WinGet needs to show more metadata about packages, specifically: size of download/disk space, date and location of package installs

### DISCUSSION THREAD

**Discussion by ObiW4nKenobi** [microsoft/winget-cli#2836] [Ideas]
I have been using WinGet for awhile now and it is a good software package manager. However, it needs to show more metadata about the packages that a user is either 1) Interested in installing or 2) has already installed.

winget search <package_name> should show the download size and required disk space for the install
  (And just in case those 2 metadata are ambiguous: download size because of compression, disk space consumed after decompression)
winget install <package_name> should, again show the download size and required disk space for the install
winget show <package_name> should show the size on disk of install, installation date and installation folder

That is all I request for now. I am a simple man. But these 3 metadata are very important and to not have them already implemented is a bewilderment to me for a package manager.
Disk space is not infinite just yet!

**Comment by Trenly** [comment]
> winget search <package_name> should show the download size and required disk space for the install
> winget install <package_name> should, again show the download size and required disk space for the install

What do you propose the technical implementation for this would look like? Winget only downloads and runs the installers just like if you had gone to the publisher's website and downloaded the exe or msi file yourself. It isn't like apt where it's downloading and de-compressing a file that contains all of the required files to run the application. From the perspective of the winget cli, there isn't a way to know how much disk space will be used after an installer is run before actually downloading and running the installer.

> winget show <package_name> should show the size on disk of install, installation date and installation folder

`winget show` is used to show details of a manifest and has nothing to do with currently installed packages. Did you mean `winget list`? If so, then winget can only access that information if the publishers write it into the registry during installation. Here's a snip from my Add/Remove Programs menu which shows that a lot of times the install location isn't known by Windows, so there isn't a way (currently) for winget to determine it.
![image](https://user-images.githubusercontent.com/12611259/211725415-e41cca4b-9108-4a4e-8f69-35e11421f9c3.png)

**Reply by ObiW4nKenobi** [reply]
If these metadata can't be determined on the fly, maybe publishing the download size/post-install disk space consumed should be a requirement to publish your packages for download. 
Even if winget cannot know the disk space required before installation it should certainly be able to calculate after the installation. Also, know the folder installation path, post install.

**Reply by Trenly** [reply]
As much as I agree, forcing developers to adopt specific best practices is very difficult. If it was easy, every publisher would be using MSI or MSIX to deliver their applications, using Semantic versioning schemes, and writing all the registry entries. This is the core challenge of winget - and since its nearly impossible to force change (consider too that some apps still work on windows even though their developers havent touched them since 2016 or maybe even earlier!), the guiding principle for this project has been to meet developers where they are. 

That said, I think it would be great if all of this data could be shown, if it can be determined. It just means that the team will need to think differently about how to detect an installation even if the program doesn’t use the best practices

**Comment by thedarkfalcon** [comment]
I really wish there was some additional metadata like _package published/updated date_, like NuGet etc have. Knowing how old a package is can be very important.

**Comment by bege10** [comment]
As winget knows the download link to the new version, checking the file size should be possible for `winget update`.

**Comment by denelon** [comment]
We have added some additional functionality related to installed packages.

Try `winget list <package> --details`

**Reply by thedarkfalcon** [reply]
Rather disappointing there is no published date included. That is one of the main things I want to see.

**Reply by denelon** [reply]
Are you looking for the date the application was published or the date it was published to the WinGet Community Repository? I assume the former. There is a field in the manifest for the date the software was released, but it's not something we're able to actually validate.

**Reply by thedarkfalcon** [reply]
As per my earlier comment, something like package published date, which NuGet has is very helpful on knowing whether the maintenance of a package has been abandoned or not. I know enforcing the application release date would be difficult, but the date the winger package was published is surely something that would be available to display?
